assets/js/modules/analytics/datastore/settings.js

```diff
@@ -55,137 +53,52 @@ export const INVARIANT_INVALID_PROFILE_NAME = 'a valid profile name is required
 export const INVARIANT_INVALID_INTERNAL_PROPERTY_ID = 'cannot submit changes with incorrect internal webPropertyID';
 export const INVARIANT_INSUFFICIENT_TAG_PERMISSIONS = 'cannot submit without proper permissions';
 
-// Actions
-const SUBMIT_CHANGES = 'SUBMIT_CHANGES';
-const START_SUBMIT_CHANGES = 'START_SUBMIT_CHANGES';
-const FINISH_SUBMIT_CHANGES = 'FINISH_SUBMIT_CHANGES';
-
-export const initialState = {
-	isDoingSubmitChanges: false,
-};
-
-export const actions = {
-	/**
-	 * Submits all changes currently present in the client, persisting them on the server.
-	 *
-	 * @since 1.9.0
-	 *
-	 * @return {Object} Empty object on success, object with `error` property on failure.
-	 */
-	*submitChanges() {
-		const registry = yield Data.commonActions.getRegistry();
-
-		yield {
-			payload: {},
-			type: START_SUBMIT_CHANGES,
-		};
-
-		const result = yield {
-			payload: {},
-			type: SUBMIT_CHANGES,
-		};
-
-		if ( result.error ) {
-			yield registry.dispatch( STORE_NAME ).receiveError( result.error, 'submitChanges', [] );
-		}
-
-		yield {
-			payload: {},
-			type: FINISH_SUBMIT_CHANGES,
-		};
+export const submitChanges = createRegistryControl( ( { select, dispatch } ) => async () => {
```

Comment:
See above comment, this shouldn't include `createRegistryControl`. Also applies elsewhere throughout the PR.

Reply:
Updated.